File: DSAlgorithms/Leetcode/c++/2_pointers/283-move_zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
       int insert_loc = 0;
       for(int iterator = 0; iterator < nums.size(); iterator++)
       {
           if(nums[iterator] != 0)
           {
               int temp = nums[insert_loc];
               nums[insert_loc] = nums[iterator];
               nums[iterator] = temp;
               insert_loc++;
           }
       }
    }
};

File: dsa_algo/leetcode/ts/src/2_pointers/283-move_zeroes.ts
/**
 Do not return anything, modify nums in-place instead.
 */
 const moveZeroes = (nums: number[]): void =>{
    let insert_loc = 0;
    for(let i = 0; i < nums.length; i++)
    {
        if(nums[i] != 0)
        {
            let temp = nums[insert_loc];
            nums[insert_loc] = nums[i];
            nums[i] = temp;
            insert_loc++;
        }
    }
};